web/src/ApiButton.tsx

```diff
@@ -307,6 +333,49 @@ async function updateButtonStatus(
   await tiltApiPut("uibuttons", "status", toUpdate)
 }
 
+function getButtonTags(button: UIButton): Tags {
+  // The location of the button in the UI
+  const component = button.spec?.location?.componentType as ApiButtonType
+
+  const buttonAnnotations = annotations(button)
+
+  // A unique hash of the button text to help identify which button was clicked
+  const specHash = buttonAnnotations[UIBUTTON_SPEC_HASH]
+
+  // Tilt-specific button annotation, currently only used to differentiate disable toggles
+  const buttonType = buttonAnnotations[UIBUTTON_ANNOTATION_TYPE]
+
+  // A toggle button will have a hidden input field with the current value of the toggle
+  // e.g., when a disable button is clicked, the hidden input will be "on" because that's
+  // the toggle's value when it's clicked, _not_ the value it's being toggled to.
+  let toggleInput: UIInputSpec | undefined
+  if (button.spec?.inputs) {
+    toggleInput = button.spec.inputs.find(
+      (input) => input.name === UIBUTTON_TOGGLE_INPUT_NAME
+    )
+  }
+
+  let toggleValue: ApiButtonToggleState | undefined
+  if (toggleInput !== undefined) {
+    const definedValue = toggleInput.hidden?.value
+    // Only use values defined in `ApiButtonToggleState`, so no user-specific information is saved.
+    // When toggle buttons are exposed in the button extension, this mini allowlist can be revisited.
+    if (
+      definedValue === ApiButtonToggleState.On ||
+      definedValue === ApiButtonToggleState.Off
+    ) {
+      toggleValue = definedValue
+    }
+  }
+
+  return {
+    buttonType,
```

Comment:
would it be better to leave off the tag completely if it's not defined, like the original code did? otherwise, i think you might end up with a bunch of {tag: "undefined"} in analytics?

Reply:
that's a great question! i believe that serializing to json drops properties that are `undefined` (`JSON.parse(JSON.stringify(tags))` does), but i'm not 100% sure of that. it's probably faster for me to make the code change than test my assumption out.

if anyone else knows, lmk for my curiosity. 🕵🏻‍♂️